Project 3/NeuralNets/src/RBFNetwork.java

```diff
@@ -0,0 +1,325 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class RBFNetwork
+{
+    private ArrayList<RBFLayer> layers = new ArrayList<>();
+
+    public RBFNetwork (int[] layerSizes, ArrayList<ArrayList<String>> condensedSet)
+    {
+
+        // create exactly 1 hidden layer for RBF network
+        RBFLayer inputLayer = new RBFLayer(layerSizes[0]);
+        RBFLayer hiddenLayer = new RBFLayer(layerSizes[1]);
+        RBFLayer outputLayer = new RBFLayer(layerSizes[2]);
+
+        hiddenLayer.setCenters(condensedSet); // set center values of our hidden layers
+
+        // set previous and next layers for each layer in an RBF
+        inputLayer.setNextLayer(hiddenLayer);
+        hiddenLayer.setPreviousLayer(inputLayer);
+        hiddenLayer.setNextLayer(outputLayer);
+        outputLayer.setPreviousLayer(hiddenLayer);
+
+        // add the three layers to the network
+        layers.add(inputLayer);
+        layers.add(hiddenLayer);
+        layers.add(outputLayer);
+
+        // initialize weights as appropriate for RBF
+        initializeWeightsRBF();
+
+        // inform each layer of its neighbors. setPreviousLayer() also updates the previous layer's nextLayer.
+        RBFLayer previousLayer = null;
```

Comment:
This was my little way of informing all the layers of their previous and next layers, even when you don't know how many hidden layers there will be. Lines 20-23 do the exact same thing, though, making this part redundant. I'd get rid of either this bit or the other bit, depending on if it's at all possible you'd ever want more hidden layers.

Reply:
Alright, will remove during cleanup since RBF will only ever have one hidden layer by definition, thanks.